### PR TITLE
Hdrp/decal ignore zero normal

### PR DESCRIPTION
**Pull request by Paulm-Unity**
### Checklist for PR maker
- Have you added a Label? : HDRP, Universal, ShaderGraph etc...
- Have you added a label for backport (if needed)? : need-backport-2019.3  .  When the PR is backported the label will be change ton backported-2019.3
- Have you added a changelog? Each package have a changelog.
- Have you updated or added the documentation for you PR? When property name is changed, when a feature behavior is change, when adding a new features, think to update the documentation in the same PR.
- Have you added a graphic test for your PR (if needed)? When adding new feature or discovering a bug that isn't cover by a test, please add a graphic test

---
### Purpose of this PR
Because decal normals are signed and stored as R8G8B8, 0 is stored as 127 and not 127.5 so if some pixels in 8x8 tile have decal normals and some don't, the ones that don't, end up incorrectly affecting material normal since they are not 0. This change checks to see if there is decal contribution per pixel and ignores pixels with no contribution within an 8x8 tile.  

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Material/Lit/LitDecalData.hlsl [hunk 1]
```diff
@@ -8,7 +8,10 @@ void ApplyDecalToSurfaceData(DecalSurfaceData decalSurfaceData, inout SurfaceDat
 
     if (decalSurfaceData.HTileMask & DBUFFERHTILEBIT_NORMAL)
     {
-        surfaceData.normalWS.xyz = normalize(surfaceData.normalWS.xyz * decalSurfaceData.normalWS.w + decalSurfaceData.normalWS.xyz);
+        if (decalSurfaceData.normalWS.w < 1.0f)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Material/Lit/LitDecalData.hlsl]: Hey, I don't understand why we require this test.
isn't it rather a precision issue?
like in this PR https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/4774? 

Note that we have a lot of location (all our shaders) where we do this thing, so if it is the correct fix, we need to do it for all materials. But I am worry about adding a test like this here

Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Material/Lit/LitDecalData.hlsl]: Hey, 
isn't this bug already fix by this PR https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/4774?
Guess the root cause of the issue is that we were not being able to encode 0 perfectly before 4774. Or it is about something else?

Note that we have a lot of location (all our shaders) where we do this thing, so if it is the correct fix, we need to do it for all materials.

Comment (Paulm-Unity) [reply com.unity.render-pipelines.high-definition/Runtime/Material/Lit/LitDecalData.hlsl]: Yes that is the same issue, I didn't know this PR actually went in, thanks!

**Comment by sebastienlagarde** [comment]
Fix already available for this.

**Pull request closed without merging**